snprc_ehr/src/client/SndLookupsManagement/components/TableGridPanel.tsx

```diff
@@ -0,0 +1,397 @@
+        setRow(currentRow['rows'][0]);
+    };
+
+    /**
+     * Initialize the queryModel for table depending on props being for parent (lookupSets) or child (lookups) table
+     */
+    const initQueryModel = async () => {
+        const baseFilters = filters;
+        actions.addModel(
+            {
+                id: table,
+                schemaQuery: schemaQuery,
+                baseFilters,
+                omittedColumns: omittedColumns,
+                bindURL: true
+            },
+            true,
+            true
+        );
+        setModelId(table);
+    };
+
+    /**
+     * Set the last selected id for the queryModel
+     */
+    const setLastSelectedId = async () => {
+        const model = queryModels[modelId];
+        if (!model || isLoading(model.selectionsLoadingState)) {
+            return;
+        }
+
+        if (model.selectionsLoadingState === LoadingState.LOADED) {
+            await updateLastSelectedId(await getLastSelectedId()).catch(error => console.error(error));
+
+        }
+    };
+
+    /**
+     * Update the state for the last selected row id in the queryModel
+     * @param id
+     */
+    const updateLastSelectedId = async (id: string) => {
+        if (selectedId !== id) {
+            if (id === undefined) {
+                actions.clearSelections(modelId);
+            } else {
+                actions.replaceSelections(modelId, [id]);
+            }
+            setSelectedId(id);
+
+        }
+    };
+
+    /**
+     * Get the row id of the most recently selected row in the queryModel
+     */
+    const getLastSelectedId = async () => {
+        const selectedIds: Set<string> = queryModels[modelId].selections;
+        return selectedIds.size > 0 ? Array.from(selectedIds).pop() : undefined;
+    };
+
+    /**
+     * Update the state for the toggleDialog value, called when a CRUD button is pressed.
+     * @param name
+     * @param requiresSelection
+     */
+    const toggleDialog = async (name: string, requiresSelection = false) => {
+        if (requiresSelection && queryModels[modelId].hasSelections) {
+            setShowDialog(undefined);
+        } else {
+            setShowDialog(name);
+        }
+    };
+
+    /**
+     * Update the state of toggleDialog to null when a CRUD modal is closed
+     */
+    const closeDialog = () =>
+    {
+        const close = async () => {
+            await toggleDialog(undefined);
+        };
+
+        try {
+            close();
+        }
+        catch(e) {
+            console.log(e);
+        }
+    }
+
+    /**
+     * Callback for completion of a create operation on a table
+     * @param response
+     */
+    const onCreateComplete = async (response: any) => {
+        closeDialog();
+        await reloadModel().catch(error => console.error(error));
+        await onRowSelectionChange(queryModels[modelId], response.rows[0][toCamelCase(rowIdName)], true, response).catch(error => console.error(error));
```

Comment:
I don't think the other two necessarily depend on the reloadModel but it might make sense to not run them if the model isn't reloaded. Going to wrap it all in a try/catch